tests/apps/courses/test_cms_views.py

```diff
@@ -224,11 +227,29 @@ def test_cms_views_autocomplete_search(self):
         # Keep language to french and search for a keyword that should match in french
         # only
         data = {"q": "picsou"}
-        payload = self.client.get(self._VIEW_URL, follow=True, data=data).json()
+        payload = self.client.get(url, follow=True, data=data).json()
         titles = [item["text"] for item in payload["results"]]
-        self.assertEqual(
-            titles,
-            [
-                "balthazar picsou",
-            ],
+        self.assertEqual(titles, ["balthazar picsou"])
+
+
+# Programmatically create test methods from the `_test_*` methods above. One set of test
+# methods for each model that has these features.
+TEST_PREFIX = "_test_cms_views_autocomplete"
+for name, related_factory in (
+    ("blogpost", factories.BlogPostFactory),
+    ("category", factories.CategoryFactory),
+    ("course", factories.CourseFactory),
+    ("organization", factories.OrganizationFactory),
+    ("person", factories.PersonFactory),
+    ("program", factories.ProgramFactory),
+):
+    for method in (
+        f for f in dir(PluginAutocompleteTestCase) if f.startswith(TEST_PREFIX)
+    ):
+        setattr(
+            PluginAutocompleteTestCase,
+            str(method).replace(TEST_PREFIX, f"{TEST_PREFIX[1:]}_{name}"),
+            lambda self, m=method, n=name, rf=related_factory: getattr(
+                PluginAutocompleteTestCase, str(m)
+            )(self, n, rf),
         )
```

Comment:
Good candidate to use `subTest` nope ?

Reply:
So you think I should rewrite it with a loop in each test method? If you think it's clearer I can do it :cry:

Reply:
I wrote a quick test and I don't know why exactly but using `subTest` with pytest do not display the context of the subTest so it is more complicated to debug the failing test than your solution. So let's keep your implementation !

Reply:
Yes. I like the fact that this code runs like real tests.